### ISSUE TITLE
There is a big problem in fluentscheduler library!!!

### ISSUE THREAD

**Issue by lampo1024** [fluentscheduler/FluentScheduler#130]
I have used "FluentScheduler" library in some of my projects,It works fine all the time.But there is a big problem in my latest winform project recently , and the whole winform application crashed. I catched the exception as follows:
```
System.InvalidOperationException: Collection was modified; enumeration operation may not execute.    
at System.Collections.Generic.HashSet`1.Enumerator.MoveNext()     
at System.Linq.Enumerable.Any[TSource](IEnumerable`1 source, Func`2 predicate)     
at FluentScheduler.JobManager.RunJob(Schedule schedule)     
at FluentScheduler.JobManager.ScheduleJobs()     
at FluentScheduler.JobManager.ScheduleJobs()    
at FluentScheduler.JobManager.ScheduleJobs()     
at FluentScheduler.JobManager.ScheduleJobs()     
at FluentScheduler.JobManager.<>c.<.cctor>b__40_0(Object state)     
at System.Threading.TimerQueueTimer.CallCallbackInContext(Object state)    
at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)    
at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)     
at System.Threading.TimerQueueTimer.CallCallback()     
at System.Threading.TimerQueueTimer.Fire()     
at System.Threading.TimerQueue.FireNextTimers()     
at System.Threading.TimerQueue.AppDomainTimerCallback()
```

**Comment by tallesl** [comment]
I don't know why you closed the issue (mistake?) but you found a pretty nasty bug alright, thank you for reporting.

Calling [`Any`] when there's a potential [`Remove`] call from another task plus [`Add`] [not being thread-safe] (as probably the [`Remove`] call). Those were really bad mistakes on my part, I apologize for that.

Looks like the fix is easy to apply (just good old `lock` for now). I'm publishing a new version ASAP.

[`Any`]: https://github.com/fluentscheduler/FluentScheduler/blob/5.1.2/Library/JobManager.cs#L388
[`Add`]: https://github.com/fluentscheduler/FluentScheduler/blob/5.1.2/Library/JobManager.cs#L443
[`Remove`]: https://github.com/fluentscheduler/FluentScheduler/blob/5.1.2/Library/JobManager.cs#L426
[not being thread-safe]: http://stackoverflow.com/a/4307283

**Comment by lampo1024** [comment]
Thank you for replying.
Please fix it ASAP , cause I'm using "FluentScheduler" in my production environment.

**Comment by cuiwenyuan** [comment]
@lampo1024 You did great job - big bug!

FluentScheduler is open sourced, this doesn't mean you can force the team to fix the bug as your expection.

**Comment by lampo1024** [comment]
@cuiwenyuan Yes, I known that.But I don't want to fix the bug by modifying the source code directly,cause It's hard to keep in step with offical version.

**Comment by tallesl** [comment]
Fixed on 5.1.3.

Sorry once again folks.
